src/main/java/com/mparticle/kits/GoogleAnalyticsFirebaseKit.java

```diff
@@ -237,16 +202,27 @@ Bundle toBundle(Map<String, String> map) {
         return bundle;
     }
 
-    List<Bundle> getProductBundles(CommerceEvent commerceEvent) {
-        List<Bundle> bundles = new ArrayList<>();
+    PickyBundle getCommerceEventBundle(CommerceEvent commerceEvent) {
+        PickyBundle pickyBundle = getTransactionAttributesBundle(commerceEvent);
+        return pickyBundle
+                .putString(FirebaseAnalytics.Param.CURRENCY, commerceEvent.getCurrency())
+                .putBundleList(FirebaseAnalytics.Param.ITEMS, getProductBundles(commerceEvent))
+                .putString(FirebaseAnalytics.Event.SET_CHECKOUT_OPTION, commerceEvent.getCheckoutOptions())
+                .putInt(FirebaseAnalytics.Event.CHECKOUT_PROGRESS, commerceEvent.getCheckoutStep());
+    }
+
+    Bundle[] getProductBundles(CommerceEvent commerceEvent) {
+        Bundle[] bundles = new Bundle[commerceEvent.getProducts().size()];
```

Comment:
this should ideally come _after_ you null-check `commerceEvent.getProducts()`. It shouldn't be null, but it definitely could be